docs/src/main/asciidoc/security-oidc-code-flow-authentication-concept.adoc

```diff
@@ -283,269 +725,152 @@ quarkus.oidc.credentials.secret=secret
+For example, when working with GitHub, the Quarkus endpoint can acquire an `access` token which will allow it to request a GitHub profile of the current user.
 
-It is required because `quarkus-oidc` expects that not only `access` and `refresh` tokens but also `IdToken` will be returned once the authorization code flow completes.
+To support the integration with such OAuth2 servers, `quarkus-oidc` needs to be configured a bit differently to allow the authorization code flow responses without `IdToken`: `quarkus.oidc.authentication.id-token-required=false`.
 
-Note, even though you will configure the extension to support the authorization code flows without `IdToken`, an internal `IdToken` will be generated to support the way `quarkus-oidc` operates where an `IdToken` is used to support the authentication session and to avoid redirecting the user to the provider such as GitHub on every request. In this case the session lifespan is set to 5 minutes which can be extended further as described in the <<session-management,session management>> section.
+[NOTE]
+--
+Even though you will configure the extension to support the authorization code flows without `IdToken`, an internal `IdToken` will be generated to standardize the way `quarkus-oidc` operates: an `IdToken` is used to support the authentication session and to avoid redirecting the user to the provider such as GitHub on every request. In this case the session lifespan is set to 5 minutes which can be extended further as described in the <<session-management,session management>> section.
 
-The next step is to ensure that the returned access token can be useful to the current Quarkus endpoint.
-If the OAuth2 provider supports the introspection endpoint then you may be able to use this access token as a source of roles with `quarkus.oidc.roles.source=accesstoken`. If no introspection endpoint is available then at the very least it should be possible to request <<user-info,UserInfo>> from this provider with `quarkus.oidc.authentication.user-info-required` - this is the case with GitHub.
+This simplifies how you handle an application supporting multiple OIDC providers.
+--
 
-Configuring the endpoint to request <<user-info,UserInfo>> is the only way `quarkus-oidc` can be integrated with the providers such as GitHub.
+The next step is to ensure that the returned access token can be useful and is valid to the current Quarkus endpoint.
+The first way is to call the OAuth2 provider introspection endpoint by configuring `quarkus.oidc.introspection-path` if the provider offers such an endpoint.
+In this case you can use the access token as a source of role using `quarkus.oidc.roles.source=accesstoken`.
+If no introspection endpoint is present, your next best step is to request <<user-info,UserInfo>> from the provider as it will at least validate the access token. Use `quarkus.oidc.token.verify-access-token-with-user-info=true`.
```

Comment:
```suggestion
If no introspection endpoint is present, you can attempt instead to request <<user-info,UserInfo>> from the provider as it will at least validate the access token. 
To do so, specify `quarkus.oidc.token.verify-access-token-with-user-info=true`.
```